ipamd/ipamd.go

```diff
@@ -746,27 +746,24 @@ func (c *IPAMContext) tryAssignIPs() (increasedPool bool, err error) {
 // 2) set up linux ENI related networking stack.
 // 3) add all ENI's secondary IP addresses to datastore
 func (c *IPAMContext) setupENI(eni string, eniMetadata awsutils.ENIMetadata) error {
-	ec2Addrs, eniPrimaryIP, err := c.getENIaddresses(eni)
```

Comment:
Note that this right here is the key to this patch. Removing the call to `getENIaddresses()` here eliminates one source of calls to DescribeNetworkInterfaces EC2 API. setupENI() is called from both `nodeInit()` *and* the reconcile loop, both of which pass in an ENIMetadata struct that has already been populated with the most up-to-date ENI information, so there is no need to call getENIAddresses() here to re-fetch that information.

Reply:
Yes, this will be called every time an ENI gets attached.